vscode-dotnet-runtime-library/src/Utils/WebRequestWorkerSingleton.ts

```diff
@@ -331,16 +332,18 @@ export class WebRequestWorkerSingleton
                     discoveredProxy = autoDetectProxies.http.toString();
                 }
             }
-            catch (error: any)
+
+            if (this.proxySettingConfiguredManually(ctx) || discoveredProxy)
```

Comment:
If you want to skip this check, you could put an "else return null" block inside the above if/else if block

Or I think you can simplify the whole thing if you just initialize discoveredProxy to what ctx?.proxyUrl then check if it's what's needed and configure it automatically if not. Then you can skip line 338.

Reply:
That's a great point, @/joeloff had some suggestions too for the last PR, so ill combine those improvements in there.